Dockerfile

```diff
@@ -1,12 +1,17 @@
-FROM cgr.dev/chainguard/go:1.19-musl as build
+FROM --platform=$BUILDPLATFORM cgr.dev/chainguard/go:1.19 as build
 
 WORKDIR /work
 
 COPY go.mod go.sum ./
 RUN go mod download
 
 COPY . .
-RUN CGO_ENABLED=0 GOOS=linux go build -v ./cmd/golink
+ARG TARGETOS TARGETARCH TARGETVARIANT
+RUN \
+    if [ "${TARGETARCH}" = "arm" ] && [ "${TARGETVARIANT}" ]; then \
```

Comment:
I'd usually expect to see a shell `-z` test in cases like this:
```
if ... && [ -z "${TARGETVARIANT}" ]; then
```

Reply:
ah, yes.  `-n` in this case, but done.